ui/src/app/pages/homepage/recent-workspaces.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+
+import {Profile, RecentWorkspace} from 'generated/fetch';
+
+import {FlexRow} from 'app/components/flex';
+import {SpinnerOverlay} from 'app/components/spinners';
+import {WorkspaceCard} from 'app/pages/workspace/workspace-card';
+import {workspacesApi} from 'app/services/swagger-fetch-clients';
+import {withUserProfile} from 'app/utils';
+
+interface State {
+  loading: boolean;
+  recentWorkspaces: RecentWorkspace[];
+}
+
+export const RecentWorkspaces = withUserProfile()
+(class extends React.Component<{ profileState: { profile: Profile, reload: Function } }, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+      recentWorkspaces: [],
+    };
+  }
+
+  componentDidMount() {
+    this.loadWorkspaces();
+  }
+
+  async loadWorkspaces() {
+    try {
+      this.setState({loading: true});
+      const recentWorkspaces = await workspacesApi().getUserRecentWorkspaces();
+      this.setState({recentWorkspaces: recentWorkspaces});
+    } catch (error) {
+      console.error(error);
```

Comment:
What error cases do we expect we might see here? I think it is within the scope of this to consider how we can display errors to the user, or if we should. If not, it is presumably being captured by the API logs, but we should figure out what gets displayed here in the workspace stead/if we want to hide it.

Reply:
Yeah, a default state might be reasonable here. 👍

Reply:
Added a 'Create new workspace' card when the section is not loading and there are no recent workspaces.